### PR TITLE
PNMA-2431: Upgrade ng-select to be compatible with Angular 17

### PR DESCRIPTION
**Pull request by ShevonDenny**
- https://cawiki.atlassian.net/browse/PNMA-2433

### CODE REVIEW

**Review by nboober — APPROVED** [review]
Looks good to me

File: package.json
Comment (geebru) [inline package.json]: Leave these out - they don't work on our end and we've removed them to avoid confusion.

**Review by geebru — CHANGES_REQUESTED** [review]
Has a local version of this been generated and imported into CAWeb as testing before we merge this?

**Comment by ShevonDenny** [comment]
> Has a local version of this been generated and imported into CAWeb as testing before we merge this?

@geebru Once the angular 17 upgrade is complete we plan to test with this version. Then we will do another release to test ng-select changes with angular 18

**Comment by geebru** [comment]
@ShevonDenny Documenting our Slack convo:
1. We can merge this into develop as is provided a build has been run and is passing and the demo has no significant functionality or UI issues.
2. For release we have two options:
    3. Do a release candidate to test the current step (A17) in CAWeb when it's ready, then the next RC at A18.
    4. Skip official RC release for 17 and just create packs to locally test, assuming we don't need an A17 version of this for any reason. Then create official RC/releases at A18 which is the target anyway.

My gut at the start was to upgrade _straight_ to A18 but the workstream on the Angular update doesn't seem like that would work in which case we can fudge the intermediary steps a bit.

Thanks Shevon!

**Review by geebru — APPROVED** [review]

**Pull request merged**
